src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonPropertyInfo.cs

```diff
@@ -135,9 +138,12 @@ public void RequiredAttributesGetDetectedAndFailDeserializationWhenValuesNotPres
             Assert.Contains(nameof(ClassWithRequiredCustomAttributes.RequiredB), exception.Message);
         }
 
-        [ConditionalFact]
+        [Fact]
         public void RequiredMemberCanBeModifiedToNonRequired()
         {
+            if (ModifiersNotSupported)
+                return;
```

Comment:
I'm not aware of any better way to succeed a test based on virtual value, please let me know if you know about it. Possibly some sort of `PassTestException` similar to `SkipTestException` but that didn't seem to exist...